### PR TITLE
Fix #2667: Add support for java date serialization

### LINKED ISSUE #2667: [Feature Request]:Java serialized support java.util.Date or java.time.LocalDateTime ...
Java serialized can't read redis value of Date or LocalDateTime ...

### PR DESCRIPTION
**Pull request by GnaneshKunal**
Fixes: #2667

image

--- 

~I raised a PR for the library repository and patched RedisInsight to work on the tip of the fix.~

~Library PR: https://github.com/asherkin/java-object-serialization/pull/1~

~By the way, the library was last updated 3 years ago.~

### CODE REVIEW

File: redisinsight/ui/src/utils/formatters/java-date.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { ObjectInputStream, JavaSerializable } from 'java-object-serialization'
+
+export default class JavaDate implements JavaSerializable {
+  time: bigint = 0n
+
+  readObject(stream: ObjectInputStream): void {
+    this.time = stream.readLong()
+  }
+
+  readResolve(): any {
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/utils/formatters/java-date.ts:10]: could you please add type here

Comment (GnaneshKunal) [reply redisinsight/ui/src/utils/formatters/java-date.ts]: I have removed the type to let the type inference kick-in.

**Review by romansergeenkosofteq — APPROVED** [review]

File: redisinsight/ui/src/utils/formatters/java-date.ts [hunk 2]
```diff
@@ -14,6 +14,23 @@ export default class JavaDate implements JavaSerializable {
   }
 
   readResolve() {
-    return new Date(Number(this.time))
+    let timeValue: number
+
+    // Handle two's complement conversion for negative numbers
+    if (this.time > 9223372036854775807n) {
+      // If the number is larger than MAX_LONG, it's a negative number in two's complement
+      timeValue = Number(this.time - 18446744073709551616n)
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/utils/formatters/java-date.ts:22]: @GnaneshKunal could you please add 9223372036854775807n and 18446744073709551616n as constant variables?

Comment (GnaneshKunal) [reply redisinsight/ui/src/utils/formatters/java-date.ts]: Done.

**Review by romansergeenkosofteq — APPROVED** [review]
lgtm. It would be nice to have more tests for dates (negative, positive, some corner cases)

**Review by mariasergeenko — APPROVED** [review]
